src/common/util.js

```diff
@@ -21,3 +21,11 @@ export const reduce = (numerator, denominator) => {
   gcd = gcd(numerator, denominator)
   return [numerator/gcd, denominator/gcd]
 }
+
+// rotates an array count times
+// taken from https://stackoverflow.com/questions/1985260/rotate-the-elements-in-an-array-in-javascript#33451102
+export const arrayRotate = (arr, count) => {
```

Comment:
If you have the points 0,1,2,3. If you rotated it, you would end up with 2,3,0,1. But the line
segment from 3,0 wasn't in the original pattern, and the segment 1,2 is broken and will no longer be
drawn.

Even though the vectors are a list of points, the pattern is really a set of line segments. This is
messing up a little by breaking the line segment between the new end and the new start.

I was thinking of building a list of line segments, and then rotating them, and then compressing
them, but I think it's probably something simpler. In fact, I think we should be able to just insert
the new final point at the very beginning.

Dealing with 0,3 is harder. That may just need to be something that the user has to deal with.

Reply:
n/m. I see you handle this with the flag to close the loop. What shapes get worse if you always insert that point? I was trying this with the wiper (lines) and it was pretty easy to get it to miss a big line.

Reply:
You are right. Rotation does not work for all shapes. It's mainly to move the ending point in certain circumstances to a place where it can connect to the next layer. It's up to you, but I'm ok with this not always working given it's a "fine tuning (advanced)" setting.